.github/workflows/pr-actions.yml

```diff
@@ -48,37 +48,61 @@ jobs:
   # Run Jest Unit Tests
   test-unit:
     uses: ./.github/workflows/run-npm-script.yml
-    needs: lint
+    needs: draft
     with:
       label: 'Test: Unit'
       script: 'npm run test:unit'
 
   # Run Cypress End to End Tests
-  test-e2e:
-    name: 'Test: End to End'
-    runs-on: ubuntu-latest
-    needs: test-unit
-    # stop the job if it runs over 20 minutes to prevent a hanging process from eating minutes
-    timeout-minutes: 45
+  install:
+    needs: draft
+    runs-on: ubuntu-22.04
     steps:
-      # Checkout and node setup are shared in both `pr-actions.yml` and `run-npm-script.yml`
-      # If you change one, please change the other
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Cypress install
+        uses: cypress-io/github-action@v5
         with:
-          node-version: '18.12.1'
-          cache: 'npm'
-      # https://github.com/cypress-io/github-action
-      - uses: cypress-io/github-action@v4
+          # Disable running of tests within install job
+          runTests: false
+          build: npm run build
+
+      - name: Save build folder
+        uses: actions/upload-artifact@v3
+        with:
+          name: build
+          if-no-files-found: error
+          path: assets
+  cypress-run:
+    runs-on: ubuntu-22.04
+    needs: install
+    strategy:
+      # don't fail the entire matrix on failure
+      fail-fast: false
```

Comment:
With `fail-fast` set to false in the workflow file:

1. All containers ran their allotted test cases and total CI runtime was ~7:30 min. 
2. The container with the failing test used the cypress-fail-fast lib to vacuously quick-pass all the tests after the failing one so essentially only the one container ends early

Reply:
Setting `fail-fast: true` did not seem to reduce the total runtime at all in case of a failure, so it probably makes sense to leave it as false and let the tests run their course

Reply:
Seems like it might be time to just remove it since the benefit won’t be as great now.